### PR TITLE
Configurable cache retention period

### PR DESCRIPTION
**Pull request by 3Rafal**
Second task from #1636

PoC:
The cache retention value was added to Mconfig. Call to `Mocaml.flush_caches` was moved to a place where Mconfig is available. Next step will be to create a flag that allows setting the config value from client. Last step is to use this flag from emacs/vim plugins.

### CODE REVIEW

**Comment by voodoos** [comment]
Is that PR ready for review ? Looks like everything is in place :-) 

Maybe we could write a test which show cache hits when the flag has a very high value and no cache hit when it is set to 0.

**Comment by 3Rafal** [comment]
@voodoos I was considering extending vim and emacs plugins, so that they can use this new flag. But this PR is actually nice and self-contained, so I'll just write tests and ask you for review when ready :)

File: src/kernel/mconfig.ml [hunk 1]
```diff
@@ -358,6 +360,14 @@ let merlin_flags = [
     marg_path (fun path merlin -> {merlin with stdlib = Some path}),
     "<path> Change path of ocaml standard library"
   );
+  (
+    "-cache-time",
+    Marg.param "int" (fun prot merlin ->
+        try {merlin with cache_time = (int_of_string prot)}
+        with _ -> invalid_arg "Valid value is int";
+      ),
+    "Change file cache retention time. Measured in minutes."
```
Comment (voodoos) [inline src/kernel/mconfig.ml:369]: Might be useful to document the default value (5) here.

**Review by voodoos — APPROVED** [review]
Looks good to me. The test is very nice :-)

File: src/kernel/mconfig.ml (hunk 1 above)
Comment (3Rafal) [reply src/kernel/mconfig.ml]: Done

**Comment by 3Rafal** [comment]
@voodoos , I've added emacs handling, because it was super simple. Here is logged result of using it (as a proof that it actually works). It's a simple query run on merlin codebase:
```
command:  ocamlmerlin server type-enclosing -protocol sexp -verbosity 0 -filename /Users/rafal/Projects/Tarides/merlin/src/kernel/mconfig.ml -cache-period 45 -position 73:3 -index 0
- result: ((assoc) (class . "return") (value ((assoc) (start (assoc) (line . 72) (col . 0)) (end (assoc) (line . 98) (col . 1)) (type . "type merlin = {
  build_path : string list;
  source_path : string list;
  cmi_path : string list;
  cmt_path : string list;
  extensions : string list;
  suffixes : (string * string) list;
  stdlib : string option;
  reader : string list;
  protocol : [ `Json | `Sexp ];
  log_file : string option;
  log_sections : string list;
  config_path : string option;
  use_ppx_cache : bool;
  exclude_query_dir : bool;
  flags_to_apply : string list with_workdir list;
  flags_applied : string list with_workdir list;
  failures : string list;
  extension_to_reader : (string * string) list;
  cache_period : int;
}") (tail . "no"))) (notifications) (timing (assoc) (clock . 60) (cpu . 37) (query . 4) (pp . 0) (reader . 4) (ppx . 0) (typer . 29) (error . 0)))
```

**Comment by voodoos** [comment]
Perfect !

Thank you !

**Pull request merged**